pkg/webhook/k8s_secrets.go

```diff
@@ -0,0 +1,112 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/flyteorg/flytestdlib/logger"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	K8sPathDefaultDirEnvVar = "FLYTE_SECRETS_DEFAULT_DIR"
+	K8sPathFilePrefixEnvVar = "FLYTE_SECRETS_FILE_PREFIX"
+	K8sSecretPathPrefix     = "/etc/flyte/secrets/"
+	K8sEnvVarPrefix         = "FLYTE_SECRETS_ENV_PREFIX"
+	K8sDefaultEnvVarPrefix  = "_FSEC_"
+	EnvVarGroupKeySeparator = "_"
+)
+
+type K8sSecretInjector struct {
+}
+
+func (i K8sSecretInjector) ID() string {
+	return "K8s"
+}
+
+func (i K8sSecretInjector) Inject(ctx context.Context, secret *core.Secret, p *corev1.Pod) (newP *corev1.Pod, injected bool, err error) {
+	switch secret.MountRequirement {
+	case core.Secret_ANY:
+		fallthrough
+	case core.Secret_FILE:
+		volumeName := secret.Key
+		if len(secret.Group) > 0 && len(secret.Key) > 0 {
+			volumeName = secret.Group + EnvVarGroupKeySeparator + secret.Key
+			p.Spec.Volumes = append(p.Spec.Volumes, corev1.Volume{
+				Name: volumeName,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: secret.Group,
+						Items: []corev1.KeyToPath{
+							{
+								Key:  secret.Key,
+								Path: secret.Key,
```

Comment:
the path should be a combination of the group and the key right?

Reply:
No this is always a "subpath" of wherever you mount this volume to in each container.